EpiAware/src/utilities.jl

```diff
@@ -28,12 +28,14 @@ function scan(f, init, xs)
 end
 
 """
-    create_discrete_pmf(dist; Δd = 1.0, D)
+    create_discrete_pmf(dist::Distribution, ::Val{:basic}; Δd = 1.0, D)
 
-Create a discrete probability mass function (PMF) from a given distribution.
+Create a discrete probability mass function (PMF) from a given distribution, assuming that the
+primary event happens at edge of an censoring inteval.
 
 Arguments:
 - `dist`: The distribution from which to create the PMF.
+- ::Val{:basic}: A dummy argument to dispatch to this method.
```

Comment:
It works, there are pitfalls at run-time if you try and leverage this kind of dispatch uncarefully e.g. [here](https://docs.julialang.org/en/v1/manual/performance-tips/#man-performance-value-type).

In this case, there is basically a good way of doing this, and a bad way we are keeping for historical/comparison purposes. And the code is nearly the same (you can't dispatch a solution method on type). So my thought was that dispatch on Val type is like a barrier. You only use this method if you _really_ want to look at single interval censoring rather than by mistake.